core/trino-main/src/main/java/io/trino/server/security/PasswordAuthenticator.java

```diff
@@ -47,22 +48,33 @@ public Identity authenticate(ContainerRequestContext request)
     {
         BasicAuthCredentials basicAuthCredentials = extractBasicAuthCredentials(request)
                 .orElseThrow(() -> needAuthentication(null));
-        try {
-            Principal principal = authenticatorManager.getAuthenticator().createAuthenticatedPrincipal(
-                    basicAuthCredentials.getUser(),
-                    basicAuthCredentials.getPassword()
-                            .orElseThrow(() -> new AuthenticationException("Malformed credentials: password is empty")));
-            String authenticatedUser = userMapping.mapUser(principal.toString());
-            return Identity.forUser(authenticatedUser)
-                    .withPrincipal(principal)
-                    .build();
-        }
-        catch (UserMappingException | AccessDeniedException e) {
-            throw needAuthentication(e.getMessage());
-        }
-        catch (RuntimeException e) {
-            throw new RuntimeException("Authentication error", e);
+        String password = basicAuthCredentials.getPassword()
+                .orElseThrow(() -> new AuthenticationException("Malformed credentials: password is empty"));
+
+        List<io.trino.spi.security.PasswordAuthenticator> authenticators = authenticatorManager.getAuthenticators();
```

Comment:
Can we keep one to one mapping between PasswordAuthenticators? So for every SPI PasswordAuthenticator you could create one PasswordAuthenticator in the engine.

Reply:
I thought about it, however this is a bigger refactor and I wanted to keep the new feature small.
Ideally we should first refactor the code and then add the feature, but that seems a lot of work right now